commcare_connect/users/models.py

```diff
@@ -13,12 +14,24 @@ class User(AbstractUser):
     check forms.SignupForm and forms.SocialSignupForms accordingly.
     """
 
+    username_validator = UnicodeUsernameValidator()
+
     # First and last name do not cover name patterns around the globe
     name = models.CharField(_("Name of User"), blank=True, max_length=255)
     first_name = None  # type: ignore
     last_name = None  # type: ignore
     email = models.EmailField(_("email address"), unique=True)
-    username = None  # type: ignore
+    username = models.CharField(
+        _("username"),
+        max_length=150,
+        unique=True,
+        help_text=_("Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only."),
+        validators=[username_validator],
+        error_messages={
+            "unique": _("A user with that username already exists."),
+        },
+        null=True,
+    )
```

Comment:
Is this just copied from `AbstractUser`? Does removing `username = None` not have the same effect?

Reply:
It is copied but it adds `null=True` since email enrolled users don't have a username.